akka-cluster-sharding/src/main/scala/akka/cluster/sharding/JoinConfigCompatCheckSharding.scala

```diff
@@ -0,0 +1,12 @@
+package akka.cluster.sharding
+
+import akka.cluster.{ ConfigValidation, JoinConfigCompatChecker }
+import com.typesafe.config.Config
+
+class JoinConfigCompatCheckSharding extends JoinConfigCompatChecker {
```

Comment:
Final so that we don't have to worry about subclassing in future changes.
Perhaps make it internal also? No reason for end users to use or see this class?